uaclient/api/u/security/package_manifest/v1.py

```diff
@@ -0,0 +1,44 @@
+from uaclient import apt, snap
+from uaclient.api.api import APIEndpoint
+from uaclient.api.data_types import AdditionalInfo
+from uaclient.config import UAConfig
+from uaclient.data_types import DataObject, Field, StringDataValue
+
+
+class InstalledPackagesResults(DataObject, AdditionalInfo):
+    fields = [
+        Field("manifest_data", StringDataValue),
+    ]
+
+    def __init__(self, manifest_data: str):
+        self.manifest_data = manifest_data
+
+
+def get_installed_packages() -> InstalledPackagesResults:
+    return _get_installed_packages(UAConfig())
+
+
+def _get_installed_packages(cfg: UAConfig) -> InstalledPackagesResults:
+    """Returns the status of installed packages (apt and snap packages)
+    Returns a string in manifest format i.e. package_name\tversion
+    """
+    manifest = ""
+    apt_pkgs = apt.get_installed_packages(include_versions=True)
+    for apt_pkg in apt_pkgs:
+        manifest += apt_pkg + "\n"
+
+    pkgs = snap.get_installed_packages()
+    for pkg in pkgs:
+        manifest += "{app}\t{version}\n".format(
+            app=pkg.name, version=pkg.version
+        )
```

Comment:
Snaps should be in the format:
```
snap:core20	stable	1587
snap:snapd	stable	16292
snap:lxd	4.0/stable/ubuntu-20.04	22753
```

which I think is `name`, `tracking`, `rev`

(example taken from https://cloud-images.ubuntu.com/releases/focal/release/ubuntu-20.04-server-cloudimg-amd64-root.manifest)

And unfortunately it looks like `snap list` cuts its output sometimes. on a fresh focal container notice the tracking channel of lxd is cut-off:
```
root@dev-f:~# snap list
Name    Version        Rev    Tracking       Publisher   Notes
core20  20220805       1611   latest/stable  canonical✓  base
lxd     4.0.9-8e2046b  22753  4.0/stable/…   canonical✓  -
snapd   2.56.2         16292  latest/stable  canonical✓  snapd
```
So we may need to run `snap info $package` for each and grab the `tracking` field from there

It is also curious that the manifest doesn't include "latest/" for the core20 or snapd channels :/

Reply:
We should probably include "latest/" instead of trying to edit what we get from snap. Oval should be able to handle both via a regex anyway